### PR TITLE
[installer] Misc improvements

### PR DESCRIPTION
**Pull request by csweichel**
## Description
This PR makes various improvements to the installer:
- clean up the import statements and group them by "stdlib", "gitopd-io", "rest of the world"
- remove dependencies to mastermind/goutils and valast
- introduce versioned config loading, including defaults
- simplify the config surface and introduce `omitempty` where appropriate

## How to test
```bash
cd installer

go run main.go init

go run main.go render --config example-config.json
```

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6023#issuecomment-933730780" title="Approved">csweichel</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/6023#issuecomment-933730780" title="Approved">csweichel</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: installer/cmd/render.go
Comment (princerachit) [inline installer/cmd/render.go:50]: how about, `config version mismatch: expected %s, got %s`. User might try newer version of config with old version of installer.

File: installer/pkg/common/common.go
Comment (princerachit) [inline installer/pkg/common/common.go:372]: Would it make sense to move this to something like `string_utils.go`? Probably other code snippets should also be moved to correctly named files.

This file is already 350+ lines and would likely keep growing. Maybe this request is out of this PRs scope but as the description says `Misc Improvements` you can consider this?

File: installer/pkg/config/v1/config.go
Comment (princerachit) [inline installer/pkg/config/v1/config.go:20]: maybe `version`? makes more readable. especially when you write `func (v v) Factory() interface{} { return &Config{} }
`

File: installer/cmd/render.go
Comment (csweichel) [reply installer/cmd/render.go:50]: Excellent point. I've updated this.

File: installer/pkg/common/common.go
Comment (csweichel) [reply installer/pkg/common/common.go:372]: A matter of taste I reckon. Usually I'd like to see a few more "string utlity" functions before introducing something like `string_utils.go`. Otherwise we'll quickly end up with a host of Go files each with one or two functions only.

File: installer/pkg/config/v1/config.go
Comment (csweichel) [reply installer/pkg/config/v1/config.go:20]: :+1:

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6023#issuecomment-933730780" title="Approved">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6023#issuecomment-934121124" title="Approved">princerachit</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/6023#issuecomment-933730780" title="Approved">csweichel</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel,princerachit]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by princerachit — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 18b6e2bec6f18ed019c92a67c5e7927d24782912</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6023#issuecomment-933730780" title="Approved">csweichel</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/6023#pullrequestreview-771044816" title="Approved">princerachit</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/6023#issuecomment-933730780" title="Approved">csweichel</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [csweichel,princerachit]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
